src/farcaster/getFarcasterUserAddress.ts

```diff
@@ -0,0 +1,34 @@
+import { getCustodyAddressForFidNeynar } from '../utils/neynar/user/getCustodyAddressForFidNeynar';
+import { getVerifiedAddressesForFidNeynar } from '../utils/neynar/user/getVerifiedAddressesForFidNeynar';
+import { FarcasterAddressType, GetFarcasterUserAddressRequest } from './types';
+
+type GetUserAddressOptions =
+  | {
+      neynarApiKey?: string;
+    }
+  | undefined;
+
+/**
+ * Get the user address for a given fid
+ * @param fid  The frame id
+ * @param FarcasterAddressType represents whether the client wants a verified address or a custody address
+ * @returns the validation addresse or the custory address. If there is an error, it returns null
+ */
+async function getFarcasterUserAddress(
+  getFarcasterUserAddressRequest: GetFarcasterUserAddressRequest,
+  options?: GetUserAddressOptions,
+): Promise<string | null> {
```

Comment:
I think we can return all addresses, and have the return type being  `Address[] | null`.
Also if this is an address we should probably use the Address type from Viem.

Reply:
Do you mean we should return both custody and verified addresses in the same call?